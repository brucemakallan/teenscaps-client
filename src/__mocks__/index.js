export const sectionInput = {
  heading1: 'test1',
  heading2: 'test2',
  heading3: 'test3',
  heading4: 'test4',
  heading5: 'test5',
  heading6: 'test6',
  body: 'test',
  dateIn: '124231221',
  dateOut: '124231222',
  parent: '312313432223',
  category: 'test',
  images: [
    'https://iviidev.info/downloads/images/sample.jpg',
  ],
  files: [
    {
      title: 'sample file',
      url: 'https://iviidev.info/downloads/files/sample.pdf',
    },
  ],
  array: [],
};

export const section = {
  ...sectionInput,
  _id: '5c9fdcf69c5e1a0d891d506e',
  dateCreated: '1553980662739',
  __v: 0,
};

export const mockState = {
  loaderReducer: { showLoader: false },
  adminReducer: { admin: {} },
  sectionsReducer: {
    sections: [section],
    section,
  },
};

export default section;
