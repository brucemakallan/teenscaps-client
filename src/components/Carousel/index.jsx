import React from 'react';
import PropTypes from 'prop-types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faLongArrowAltLeft, faLongArrowAltRight } from '@fortawesome/free-solid-svg-icons';

const carouselId = id => `imageCarousel-${id}`;

const Carousel = ({ id, imageUrls, captions }) => (
  <div id={carouselId(id)} className="carousel slide" data-ride="carousel">
    <ol className="carousel-indicators">
      {imageUrls.map((url, index) => (
        <li
          data-target={`#${carouselId(id)}`}
          data-slide-to={index.toString()}
          className={index === 0 ? 'active' : ''}
          key={url}
        />
      ))}
    </ol>

    <div className="carousel-inner">
      {imageUrls.map((url, index) => (
        <div className={index === 0 ? 'carousel-item active' : 'carousel-item'} key={url}>
          <img className="d-block h-100" src={url} alt="poster" />
          {captions && captions[index] && (
            <div className="carousel-caption d-none d-md-block">
              <span className="caption-heading">{captions[index]}</span>
            </div>
          )}
        </div>
      ))}
    </div>

    <a className="carousel-control-prev nextPrevIcon" href={`#${carouselId(id)}`} role="button" data-slide="prev">
      <FontAwesomeIcon className="large" icon={faLongArrowAltLeft} />
      <span className="sr-only">Previous</span>
    </a>
    <a className="carousel-control-next nextPrevIcon" href={`#${carouselId(id)}`} role="button" data-slide="next">
      <FontAwesomeIcon className="large" icon={faLongArrowAltRight} />
      <span className="sr-only">Next</span>
    </a>
  </div>
);

Carousel.propTypes = {
  id: PropTypes.string.isRequired,
  imageUrls: PropTypes.arrayOf(PropTypes.string).isRequired,
  captions: PropTypes.arrayOf(PropTypes.string),
};

Carousel.defaultProps = {
  captions: []
};

export default Carousel;
