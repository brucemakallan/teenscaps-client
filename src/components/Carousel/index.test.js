import React from 'react';
import { mount } from 'enzyme';
import Carousel from '.';
import { section } from '../../__mocks__';

describe('Carousel', () => {
  it('should render correctly', () => {
    const wrapper = mount(<Carousel id="sample-id" imageUrls={section.images} />);
    expect(wrapper).toMatchSnapshot();
  });
});
