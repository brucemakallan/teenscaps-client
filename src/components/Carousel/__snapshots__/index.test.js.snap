// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel should render correctly 1`] = `
<Carousel
  captions={Array []}
  id="sample-id"
  imageUrls={
    Array [
      "https://iviidev.info/downloads/images/sample.jpg",
    ]
  }
>
  <div
    className="carousel slide"
    data-ride="carousel"
    id="imageCarousel-sample-id"
  >
    <ol
      className="carousel-indicators"
    >
      <li
        className="active"
        data-slide-to="0"
        data-target="#imageCarousel-sample-id"
        key="https://iviidev.info/downloads/images/sample.jpg"
      />
    </ol>
    <div
      className="carousel-inner"
    >
      <div
        className="carousel-item active"
        key="https://iviidev.info/downloads/images/sample.jpg"
      >
        <img
          alt="poster"
          className="d-block h-100"
          src="https://iviidev.info/downloads/images/sample.jpg"
        />
      </div>
    </div>
    <a
      className="carousel-control-prev nextPrevIcon"
      data-slide="prev"
      href="#imageCarousel-sample-id"
      role="button"
    >
      <FontAwesomeIcon
        border={false}
        className="large"
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              448,
              512,
              Array [],
              "f30a",
              "M134.059 296H436c6.627 0 12-5.373 12-12v-56c0-6.627-5.373-12-12-12H134.059v-46.059c0-21.382-25.851-32.09-40.971-16.971L7.029 239.029c-9.373 9.373-9.373 24.569 0 33.941l86.059 86.059c15.119 15.119 40.971 4.411 40.971-16.971V296z",
            ],
            "iconName": "long-arrow-alt-left",
            "prefix": "fas",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        symbol={false}
        title=""
        transform={null}
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-long-arrow-alt-left fa-w-14 large"
          data-icon="long-arrow-alt-left"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M134.059 296H436c6.627 0 12-5.373 12-12v-56c0-6.627-5.373-12-12-12H134.059v-46.059c0-21.382-25.851-32.09-40.971-16.971L7.029 239.029c-9.373 9.373-9.373 24.569 0 33.941l86.059 86.059c15.119 15.119 40.971 4.411 40.971-16.971V296z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </FontAwesomeIcon>
      <span
        className="sr-only"
      >
        Previous
      </span>
    </a>
    <a
      className="carousel-control-next nextPrevIcon"
      data-slide="next"
      href="#imageCarousel-sample-id"
      role="button"
    >
      <FontAwesomeIcon
        border={false}
        className="large"
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              448,
              512,
              Array [],
              "f30b",
              "M313.941 216H12c-6.627 0-12 5.373-12 12v56c0 6.627 5.373 12 12 12h301.941v46.059c0 21.382 25.851 32.09 40.971 16.971l86.059-86.059c9.373-9.373 9.373-24.569 0-33.941l-86.059-86.059c-15.119-15.119-40.971-4.411-40.971 16.971V216z",
            ],
            "iconName": "long-arrow-alt-right",
            "prefix": "fas",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        symbol={false}
        title=""
        transform={null}
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-long-arrow-alt-right fa-w-14 large"
          data-icon="long-arrow-alt-right"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M313.941 216H12c-6.627 0-12 5.373-12 12v56c0 6.627 5.373 12 12 12h301.941v46.059c0 21.382 25.851 32.09 40.971 16.971l86.059-86.059c9.373-9.373 9.373-24.569 0-33.941l-86.059-86.059c-15.119-15.119-40.971-4.411-40.971 16.971V216z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </FontAwesomeIcon>
      <span
        className="sr-only"
      >
        Next
      </span>
    </a>
  </div>
</Carousel>
`;
